src/NServiceBus.Core/Unicast/Messages/MessageMetadataRegistry.cs

```diff
@@ -69,7 +69,18 @@ internal IEnumerable<MessageMetadata> GetAllMessages()
             return new List<MessageMetadata>(messages.Values);
         }
 
-        internal MessageMetadata RegisterMessageType(Type messageType)
+        internal void RegisterAllMessageTypes(IList<Type> availableTypes)
+        {
+            foreach (var type in availableTypes)
+            {
+                if (conventions.IsMessageType(type))
```

Comment:
Btw. I move the convention checking for messages into the message metadata registry itself. Which makes sense in my point of view because previously that was done from the serializer feature itself. Of course this influence the following test https://github.com/Particular/NServiceBus/pull/3948/files#diff-b3add5fea583ba23ddc63551439df777R23 but I think we are safe since the registry throws anyway when to try to retrieve metadata over GetMessageMetadata and conventions.IsMessageType fails.